lib/simulateSpendCondTxLocally.js

```diff
@@ -0,0 +1,289 @@
+  Object.keys(erc20sToBeUsed).forEach(item => {
+    colorAddrMap[item] = Buffer.from(erc20sToBeUsed[item].address.replace('0x', ''), 'hex')
+  })
+
+  const inputMap = {};
+  for (let i = 0; i < tx.inputs.length; i += 1) {
+    inputMap[tx.inputs[i].prevout.getUtxoId()] =
+      unspent[tx.inputs[i].prevout.hex()];
+  }
+
+  // creating a new VM instance
+  const vm = new VM({
+    hardfork: 'petersburg'
+  });
+
+  // creating the reactor account with some wei
+  const reactorAccount = new Account();
+  reactorAccount.balance = '0xf00000000000000001';
+  await setAccount(reactorAccount, REACTOR_ADDR, vm.stateManager);
+
+  let nonceCounter = 0;
+
+  // deploy erc20 contracts used in this simulation
+  // eslint-disable-next-line  guard-for-in
+  for (const color in Object.keys(colorAddrMap)) {
+    // eslint-disable-next-line no-await-in-loop
+    await setAccountCode(erc20sToBeUsed[color].erc20code || minErc20Code, colorAddrMap[color], vm.stateManager); // eslint-disable-line no-await-in-loop
+  }
+
+  // mint tokens according to unspent
+  // eslint-disable-next-line guard-for-in
+  for (const key in unspent) {
+    // minting amount of output to address of condition
+    const unspentItem = unspent[key]
+    const amountHex = utils.setLengthLeft(
+      utils.toBuffer(`0x${BigInt(unspentItem.value).toString(16)}`),
+      32
+    );
+    const addr = Buffer.from(
+      unspentItem.address.replace('0x', ''),
+      'hex'
+    )
+    // eslint-disable-next-line no-await-in-loop
+    await runTx(vm, {
+      nonce: nonceCounter,
+      gasLimit: BLOCK_GAS_LIMIT,
+      to: colorAddrMap[unspentItem.color],
+      data: Buffer.concat([MINT_FUNCSIG, addr, amountHex]),
+    });
+    nonceCounter += 1;
+  }
+
+  // deploying conditions
+  tx.inputs.forEach(async input => {
+    // ripemd160 of SpendCond as SC contract address
+    await setAccountCode(input.script, utils.ripemd160(input.script), vm.stateManager);
+  });
+
+  // need to commit to trie, needs a checkpoint first 
+  await new Promise((resolve) => {
+    vm.stateManager.checkpoint(() => {
+      vm.stateManager.commit(() => {
+        resolve();
+      });
+    });
+  });
+
+  const logOuts = [];
+  const usedGasAggrByColor = {};
+  // running conditions with msgData
+  for (let i = 0; i < tx.inputs.length; i += 1) {
+    // eslint-disable-next-line no-await-in-loop
+    const results = await runTx(vm, {
+      nonce: nonceCounter,
+      gasLimit: BLOCK_GAS_LIMIT,
+      to: utils.ripemd160(tx.inputs[i].script),
+      data: tx.inputs[i].msgData,
+    });
+    nonceCounter += 1;
+    const out = inputMap[tx.inputs[i].prevout.getUtxoId()];
+    if (usedGasAggrByColor[out.color]) {
+      usedGasAggrByColor[out.color] = add(
+        usedGasAggrByColor[out.color],
+        BigInt(results.gasUsed)
+      );
+    } else {
+      usedGasAggrByColor[out.color] = BigInt(results.gasUsed);
+    }
+    // iterate through all transfer events and get outs
+    results.vm.logs.forEach(log => {
```

Comment:
could it be other event here apart from Transfer?

Reply:
@troggy 
good point.
currently we can add a basic transfer event structure check